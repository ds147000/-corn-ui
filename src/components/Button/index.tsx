import React, { useMemo } from 'react'
import ClassNames from 'classnames'
import { Button as TaroButton } from '@tarojs/components'
import Link, { LinkProps } from '../Link'
import { ITouchEvent } from '../../types'

/** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
export type ButtonSize = 'max' | 'big' | 'large' | 'middle' | 'small' | 'mini'
export type ButtonType = 'primary' | 'warn' | 'error' | 'link' | 'pop' | 'default' | 'light' | 'stop'

export interface ButtonProps extends LinkProps {
  /** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
  size?: ButtonSize
  /** 按钮风格类型 */
  type?: ButtonType
  /** 图标 */
  icon?: React.ReactNode
  /** 将按钮宽度调整为其父宽度的选项	 */
  block?: boolean
  /** 镂空风格 */
  ghost?: boolean
  /** 是否禁用 */
  disabled?: boolean
  /** 跳转链接，填写启用A链接行为 */
  to?: string
  /** 表单点击行为 */
  formType?: 'submit' | 'reset'
  /** 开启固定宽度 */
  auto?: boolean
  /** 点击事件 */
  onClick?(e: ITouchEvent): void
}

const Button: React.FC<ButtonProps> = ({
  size = 'middle',
  type = 'primary',
  block,
  ghost,
  disabled,
  to,
  onClick,
  icon,
  children,
  className,
  formType,
  auto,
  ...props
}) => {
  const isIcon = icon !== undefined
  const isChildren = children !== undefined

  const _class = useMemo(() => {
    return ClassNames(
      'corn-btn',
      'corn-if corn-ac corn-jc',
      `corn-btn-${size}`,
      `corn-btn-${type}`,
      {
        'corn-btn-block': block,
        'corn-btn-ghost': ghost,
        'corn-btn-icon': isIcon && !isChildren,
        'corn-btn-child-icon': isIcon && isChildren,
        'corn-btn-disabled': disabled,
        'corn-btn-auto': auto
      },
      className
    )
  }, [ size, type, block, ghost, disabled, className, auto, isIcon, isChildren ])

  const _onClick = (e: ITouchEvent): void => {
    if (disabled) return

    onClick?.(e)
  }



  if (to) {
    return (
      <Link {...props} target="Text" to={to} className={_class} onClick={_onClick} disabled={disabled} type="default" >
        {children}
        {icon}
      </Link>
    )
  }

  return (
    <TaroButton {...props} formType={formType} className={_class} onClick={_onClick} >
      {children}
      {icon}
    </TaroButton>
  )
}

export default Button
