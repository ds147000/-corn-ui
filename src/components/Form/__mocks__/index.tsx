import Form from '../index.h5'

export default Form
