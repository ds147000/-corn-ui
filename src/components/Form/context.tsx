/* eslint-disable no-empty-function, @typescript-eslint/no-empty-function */
import { createContext } from 'react'

export type SetValue = (value?: unknown) => void

export type FormContextValue = {
  add(name: string, cb: SetValue): void
  remove(name: string): void
}

const formContextValue: FormContextValue = {
  add(_name: string, _resCb: SetValue): void {
  },
  remove(_name): void {
  }
}

export const FromContext = createContext(formContextValue)
