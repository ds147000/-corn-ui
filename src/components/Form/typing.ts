/* eslint-disable no-magic-numbers */

export interface FromProps {
  /** 默认表单值 */
  defaultValue?: Record<string, unknown>
  /** 提交事件 */
  onSubmit?(result: Record<string, unknown>): void
  /** 重置事件 */
  onReset?(): void
}



