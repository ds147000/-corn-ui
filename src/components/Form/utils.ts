export function transfromInputValue(value: string): unknown {
  if (!value) return value

  try {
    return JSON.parse(value)
  } catch (error) {
    return value
  }
}
