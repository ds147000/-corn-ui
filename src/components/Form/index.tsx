/* eslint-disable import/export */
/* eslint-disable import/first */
// #if _APP === 'weapp'
import Form from './index.taro'

export default Form
// #else
import Form from './index.h5'

export default Form
// #endif
