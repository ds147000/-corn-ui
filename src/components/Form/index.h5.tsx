import React from 'react'
import { FromContext, SetValue } from './context'
import { FormClass, FormProps } from './typing'
import { transfromInputValue } from './utils'

const INIT_TIMER = 60

class Form extends React.Component<FormProps> implements FormClass {
  private el: HTMLFormElement
  private setMap = new Map<string, SetValue>()
  private timeId: NodeJS.Timeout

  getValue = async (): Promise<Record<string, unknown>> => {
    const result: Record<string, unknown> = {}

    this.el.querySelectorAll('input').forEach((item) => {
      if (item.name) result[item.name] = transfromInputValue(item.value)
    })

    this.el.querySelectorAll('textarea').forEach((item) => {
      if (item.name) result[item.name] = transfromInputValue(item.value)
    })

    return result
  }

  setValue = (data: Record<string, unknown>): void => {
    Object.keys(data).map((key) => {
      if (this.setMap.has(key)) {
        this.setMap.get(key)?.(data[key])
        return
      }
    })
  }

  submit = (): void => {
    this.getValue()
      .then((res) => this.props.onSubmit?.(res))
  }

  reset = (): void => {
    this._onReset()
  }


  private _onSubmit = async (event: React.FormEvent<HTMLFormElement>): Promise<void> => {
    event.preventDefault()

    const data = await this.getValue()
    this.props.onSubmit?.(data)
  }

  private _onReset = (): void => {
    this.setMap.forEach((item) => item())

    this.props.onReset?.()
  }

  private _putResetCbList = (name: string, set: SetValue): void => {
    this.setMap.set(name, set)
  }

  private _removeResetCbList = (name: string): void => {
    this.setMap.delete(name)
  }

  componentDidMount(): void {
    this.timeId = setTimeout(() => { // 延迟执行，防止子组件未完成渲染
      if (this.props.defaultValue) this.setValue(this.props.defaultValue)
    }, INIT_TIMER)
  }

  componentWillUnmount(): void {
    clearTimeout(this.timeId)
  }

  render(): JSX.Element {
    return (
      <FromContext.Provider value={{ add: this._putResetCbList, remove: this._removeResetCbList }}>
        {/* eslint-disable-next-line react/forbid-elements */}
        <form
          onSubmit={this._onSubmit}
          onReset={this._onReset}
          ref={(ref): HTMLFormElement => this.el = ref as HTMLFormElement}
        >
          {this.props.children}
        </form>
      </FromContext.Provider>
    )
  }
}

export default Form
