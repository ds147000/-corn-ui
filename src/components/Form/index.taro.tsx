import React from 'react'
import { BaseEventOrig, Form as TaroForm } from '@tarojs/components'
import { FormProps as TaroFormProps } from '@tarojs/components/types/Form'
import { FromProps } from './typing'
import { FromContext, SetValue } from './context'
import { transfromInputValue } from './utils'

class Form extends React.Component<FromProps> {
  private el = React.createRef<{ uid: string }>()
  private setMap = new Map<string, SetValue>()

  setValue = (data: Record<string, unknown> = {}): void => {
    Object.keys(data).map((key) => {
      if (this.setMap.has(key)) {
        this.setMap.get(key)?.(data[key])
        return
      }

      // const input = this.el.querySelector(`input[name='${key}']`) as HTMLInputElement
      // input?.setAttribute('value', String(data[key]))

    })
  }

  componentDidMount(): void {
    if (this.props.defaultValue) this.setValue(this.props.defaultValue)
  }

  private _onSubmit = (event: BaseEventOrig<TaroFormProps.onSubmitEventDetail>): void => {
    const data: Record<string, unknown> = {}
    Object.keys(event.detail.value || {})
      .map((key) => data[key] = transfromInputValue((event.detail.value as Record<string, string>)[key]))

    this.props.onSubmit?.(data)
  }

  private _onReset = (): void => {
    this.setValue()
    this.props.onReset?.()
  }

  private _putResetCbList = (name: string, set: SetValue): void => {
    this.setMap.set(name, set)
  }

  private _removeResetCbList = (name: string): void => {
    this.setMap.delete(name)
  }

  render(): JSX.Element {
    return (
      <FromContext.Provider value={{ add: this._putResetCbList, remove: this._removeResetCbList }}>
        <TaroForm
          onSubmit={this._onSubmit}
          onReset={this._onReset}
          ref={this.el}
        >
          {this.props.children}
        </TaroForm>
      </FromContext.Provider>
    )
  }
}

export default Form
