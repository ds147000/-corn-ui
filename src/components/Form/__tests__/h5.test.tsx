import React, { useEffect, useState } from 'react'
import { act, render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Form from '../index.h5'
import Input from '../../Input'
import Checkbox from '../../Checkbox/checkbox'
import CheckboxGroup from '../../Checkbox/group'
import Button from '../../Button'
import Textarea from '../../Textarea'

beforeAll(() => {
  jest.useFakeTimers()
})

afterAll(() => {
  jest.useRealTimers()
})

describe('Fomr', () => {
  test('渲染', async () => {
    const screen = render(
      <Form>
        <Input />
        <Checkbox />
        <Textarea />
      </Form>
    )

    expect(screen.container.getElementsByTagName('input').length).toBe(2)

    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('自带值', async () => {
    const screen = render(
      <Form>
        <Input value="123" />
        <Checkbox check>男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">1</Checkbox>
          <Checkbox value="2">2</Checkbox>
          <Checkbox value="3">3</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    expect(input.value).toBe("123")

    const check = screen.getByText('男').querySelector('input') as HTMLInputElement
    expect(check.value).toBe("true")

    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('初始化值', async () => {
    const screen = render(
      <Form defaultValue={{ name: '小明', sex: false, liuyan: '123456', abc: '122' }} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">1</Checkbox>
          <Checkbox value="2">2</Checkbox>
          <Checkbox value="3">3</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
      </Form>
    )
    await waitFor(() => expect((screen.getByTestId('input') as HTMLInputElement).value)
      .toBe("小明"))
    await waitFor(() => expect((screen.getByText('男').querySelector('input') as HTMLInputElement).value)
      .toBe("false"))

    await waitFor(() => expect((screen.getByTestId('textarea') as HTMLTextAreaElement).value)
      .toBe('123456'))

    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('改变初始化值', async () => {
    const App = () => {
      const [value, setValue] = useState({ name: '小明', sex: false, liuyan: "123" })

      useEffect(() => {
        setTimeout(() => setValue({ name: '123', sex: true, liuyan: "456" }))
      },[])

      return (
        <Form defaultValue={value} >
          <Input name="name" />
          <Checkbox name="sex">男</Checkbox>
          <CheckboxGroup name="age" >
            <Checkbox value="1">1</Checkbox>
            <Checkbox value="2">2</Checkbox>
            <Checkbox value="3">3</Checkbox>
          </CheckboxGroup>
          <CheckboxGroup name="study" radio >
            <Checkbox value="1">大学</Checkbox>
            <Checkbox value="2">小学</Checkbox>
          </CheckboxGroup>
          <Textarea name="liuyan" />
        </Form>
      )
    }

    const screen = render(<App />)

    const input = screen.getByTestId('input') as HTMLInputElement
    await waitFor(() => expect(input.value).toBe("123"))

    const check = screen.getByText('男').querySelector('input') as HTMLInputElement
    await waitFor(() => expect(check.value).toBe("true"))

    const textarea = screen.getByTestId('textarea') as HTMLTextAreaElement
    await waitFor(() => expect(textarea.value).toBe('456'))
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('提交', async () => {
    const onSubmit = jest.fn()

    const screen = render(
      <Form onSubmit={onSubmit} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
        <Button formType="submit" >提交</Button>
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('100'))
    userEvent.click(screen.getByText('大学'))
    userEvent.type(screen.getByTestId('textarea'), '123456')
    userEvent.click(screen.getByText('提交'))

    await waitFor(() => expect(onSubmit.mock.calls[0][0]).toEqual({ name: 'hello wrold', sex: true, age: ['1'], study: '1', liuyan: 123456 }))
  })

  test('具有初始化值的提交', async () => {
    const onSubmit = jest.fn()

    const screen = render(
      <Form onSubmit={onSubmit} defaultValue={{ name: '1', sex: false, liuyan: '123456' }} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
        <Button formType="submit" >提交</Button>
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('100'))
    userEvent.click(screen.getByText('大学'))
    userEvent.type(screen.getByTestId('textarea'), '00000')
    userEvent.click(screen.getByText('提交'))
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ name: 'hello wrold', sex: true, age: ['1'], study: '1', liuyan: '00000' }))
  })

  test('自管理提交', async () => {
    const onSubmit = jest.fn()

    const screen = render(
      <Form onSubmit={onSubmit} >
        <Input name="name" value="1234567" onInput={() => {}} />
        <Checkbox name="sex" check={false} >男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
        <Button formType="submit" >提交</Button>
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('200'))
    userEvent.click(screen.getByText('大学'))
    userEvent.type(screen.getByTestId('textarea'), '我是多行文本')
    userEvent.click(screen.getByText('提交'))
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ name: 1234567, sex: false, age: ['2'], study: '1', liuyan: '我是多行文本' }))
  })

  test('重置', async () => {
    const onSubmit = jest.fn()

    const screen = render(
      <Form onSubmit={onSubmit} defaultValue={{ name: '1', sex: false }} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
        <Button formType="submit" >提交</Button>
        <Button formType="reset" >重置</Button>
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('200'))
    userEvent.click(screen.getByText('大学'))
    userEvent.type(screen.getByTestId('textarea'), '我是多行文本')
    userEvent.click(screen.getByText('重置'))
    userEvent.click(screen.getByText('提交'))
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ name: '', sex: false, age: [], study: '', liuyan: '' }))
  })

  test('实例使用-提交', async () => {
    const onSubmit = jest.fn()
    const form = React.createRef<Form>()

    const screen = render(
      <Form onSubmit={onSubmit} ref={form} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('200'))
    userEvent.click(screen.getByText('大学'))
    userEvent.type(screen.getByTestId('textarea'), '我是多行文本')
    form.current.submit()
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ name: 'hello wrold', sex: true, age: ['2'], study: '1', liuyan: '我是多行文本' }))
  })

  test('实例使用-重置', async () => {
    const onSubmit = jest.fn()
    const form = React.createRef<Form>()

    const screen = render(
      <Form onSubmit={onSubmit} ref={form} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    userEvent.click(screen.getByText('200'))
    userEvent.click(screen.getByText('小学'))
    userEvent.type(screen.getByTestId('textarea'), 'abcd')
    form.current.reset()
    form.current.submit()
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ sex: false, name: '', age: [], study: '', liuyan: '' }))
  })

  test('实例使用-获取值', async () => {
    const onSubmit = jest.fn()
    const form = React.createRef<Form>()

    const screen = render(
      <Form onSubmit={onSubmit} ref={form} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
        <Textarea />
      </Form>
    )

    const input = screen.getByTestId('input') as HTMLInputElement
    userEvent.type(input, 'hello wrold')

    const check = screen.getByText('男') as HTMLInputElement
    userEvent.click(check)
    await (await screen.findAllByTestId('textarea')).map((item) => {
      userEvent.type(item, '1234567')
    })
    expect(await form.current.getValue())
      .toEqual({ name: 'hello wrold', sex: true, age: [], study: '', liuyan: 1234567 })
  })

  test('实例使用-设置值', async () => {
    const onSubmit = jest.fn()
    const form = React.createRef<Form>()

    render(
      <Form onSubmit={onSubmit} ref={form} >
        <Input name="name" />
        <Checkbox name="sex">男</Checkbox>
        <CheckboxGroup name="age" >
          <Checkbox value="1">100</Checkbox>
          <Checkbox value="2">200</Checkbox>
          <Checkbox value="3">300</Checkbox>
        </CheckboxGroup>
        <CheckboxGroup name="study" radio >
          <Checkbox value="1">大学</Checkbox>
          <Checkbox value="2">小学</Checkbox>
        </CheckboxGroup>
        <Textarea name="liuyan" />
      </Form>
    )
    form.current.setValue({ name: 'hello wrold', sex: true, age: ['2'], study: '1', liuyan: '我是多行文本'})
    form.current.submit()
    await waitFor(() => expect(onSubmit.mock.calls[0][0])
      .toEqual({ name: 'hello wrold', sex: true, age: ['2'], study: '1', liuyan: '我是多行文本' }))
  })
})
