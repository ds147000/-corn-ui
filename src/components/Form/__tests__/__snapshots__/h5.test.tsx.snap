// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fomr 初始化值 1`] = `
<div>
  <form>
    <input
      class="weui-input xrk-input"
      data-testid="input"
      name="name"
      type="text"
      value="小明"
    />
    <div
      class="xrk-if xrk-ac xrk-checkbox"
      data-testid="xrk-checkbox"
    >
      <input
        class="xrk-checkbox-hide"
        data-testid="check"
        name="sex"
        readonly=""
        value="false"
      />
      <i
        class="xrk-icon xrk-i-radio"
      />
      男
    </div>
    <div
      class=""
      name="age"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="age"
        readonly=""
        value="[]"
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        1
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        2
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="3"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        3
      </div>
    </div>
    <div
      class=""
      name="study"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="study"
        readonly=""
        value=""
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        大学
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        小学
      </div>
    </div>
    <div
      class="xrk-textarea"
    >
      <textarea
        class="xrk-textarea-conter"
        data-testid="textarea"
        maxlength="300"
        name="liuyan"
        placeholder=""
      >
        123456
      </textarea>
      <div
        class="xrk-textarea-number"
      >
        6
        /
        300
      </div>
      <div
        class="xrk-textarea-floor"
      />
    </div>
  </form>
</div>
`;

exports[`Fomr 改变初始化值 1`] = `
<div>
  <form>
    <input
      class="weui-input xrk-input"
      data-testid="input"
      name="name"
      type="text"
      value="123"
    />
    <div
      class="xrk-if xrk-ac xrk-checkbox-active xrk-checkbox"
      data-testid="xrk-checkbox"
    >
      <input
        class="xrk-checkbox-hide"
        data-testid="check"
        name="sex"
        readonly=""
        value="true"
      />
      <i
        class="xrk-icon xrk-i-fill-select"
      />
      男
    </div>
    <div
      class=""
      name="age"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="age"
        readonly=""
        value="[]"
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        1
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        2
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="3"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        3
      </div>
    </div>
    <div
      class=""
      name="study"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="study"
        readonly=""
        value=""
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        大学
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        小学
      </div>
    </div>
    <div
      class="xrk-textarea"
    >
      <textarea
        class="xrk-textarea-conter"
        data-testid="textarea"
        maxlength="300"
        name="liuyan"
        placeholder=""
      >
        456
      </textarea>
      <div
        class="xrk-textarea-number"
      >
        3
        /
        300
      </div>
      <div
        class="xrk-textarea-floor"
      />
    </div>
  </form>
</div>
`;

exports[`Fomr 渲染 1`] = `
<div>
  <form>
    <input
      class="weui-input xrk-input"
      data-testid="input"
      type="text"
      value=""
    />
    <div
      class="xrk-if xrk-ac xrk-checkbox"
      data-testid="xrk-checkbox"
    >
      <input
        class="xrk-checkbox-hide"
        data-testid="check"
        readonly=""
        value="false"
      />
      <i
        class="xrk-icon xrk-i-radio"
      />
    </div>
    <div
      class="xrk-textarea"
    >
      <textarea
        class="xrk-textarea-conter"
        data-testid="textarea"
        maxlength="300"
        placeholder=""
      />
      <div
        class="xrk-textarea-number"
      >
        0
        /
        300
      </div>
      <div
        class="xrk-textarea-floor"
      />
    </div>
  </form>
</div>
`;

exports[`Fomr 自带值 1`] = `
<div>
  <form>
    <input
      class="weui-input xrk-input"
      data-testid="input"
      type="text"
      value="123"
    />
    <div
      class="xrk-if xrk-ac xrk-checkbox-active xrk-checkbox"
      data-testid="xrk-checkbox"
    >
      <input
        class="xrk-checkbox-hide"
        data-testid="check"
        readonly=""
        value="true"
      />
      <i
        class="xrk-icon xrk-i-fill-select"
      />
      男
    </div>
    <div
      class=""
      name="age"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="age"
        readonly=""
        value="[]"
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        1
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        2
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="3"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        3
      </div>
    </div>
    <div
      class=""
      name="study"
    >
      <input
        class="xrk-checkbox-group-hide"
        data-testid="check"
        name="study"
        readonly=""
        value=""
      />
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="1"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        大学
      </div>
      <div
        class="xrk-if xrk-ac xrk-checkbox"
        data-testid="xrk-checkbox"
      >
        <input
          class="xrk-checkbox-hide"
          data-testid="check"
          name=""
          readonly=""
          value="2"
        />
        <i
          class="xrk-icon xrk-i-radio"
        />
        小学
      </div>
    </div>
    <div
      class="xrk-textarea"
    >
      <textarea
        class="xrk-textarea-conter"
        data-testid="textarea"
        maxlength="300"
        name="liuyan"
        placeholder=""
      />
      <div
        class="xrk-textarea-number"
      >
        0
        /
        300
      </div>
      <div
        class="xrk-textarea-floor"
      />
    </div>
  </form>
</div>
`;
